src/Symfony/Component/Form/FormRegistry.php

```diff
@@ -82,11 +82,14 @@ public function getType($name)
 
             if (!$type) {
                 // Support fully-qualified class names
-                if (class_exists($name) && in_array('Symfony\Component\Form\FormTypeInterface', class_implements($name))) {
-                    $type = new $name();
-                } else {
-                    throw new InvalidArgumentException(sprintf('Could not load type "%s"', $name));
+                if (!class_exists($name)) {
+                  throw new InvalidArgumentException(sprintf('Could not load type "%s": class does not exist', $name));
                 }
+                else if (!in_array('Symfony\Component\Form\FormTypeInterface', class_implements($name))) {
+                  throw new InvalidArgumentException(sprintf('Could not load type "%s": class does not implement "Symfony\Component\Form\FormTypeInterface"', $name));
```

Comment:
instead of manually writing the interface name we can now make use of the `class` constant (same above)